### PR TITLE
Feat/manager 9189: configuration page implementation

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `MANAGER-9069`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-9189
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
1. configuration page implementation
2.1 step 1: name archive
2.2 step 2: link user to archive

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js [hunk 1]
```diff
@@ -39,7 +72,63 @@ export default class ColdArchiveConfigurationController {
     );
   }
 
+  isReadyForValidation() {
+    const { createOrLinkedMode } = this.userModel;
+    const { createMode, linkedMode } = this.userModel;
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js:77]: **nitpick:** Two statements can be combined
```suggestion
    const { createMode, linkedMode, createOrLinkedMode } = this.userModel;
```

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        ? `${user.username} - ${user.description}`
+        : user.username,
+    }));
+  }
+
+  getCredentialTranslation(user) {
+    return this.$translate.instant(
+      user.s3Credentials.length > 0
+        ? 'pci_projects_project_storages_cold_archive_add_step_link_user_archive_mode_select_list_has_credential'
+        : 'pci_projects_project_storages_cold_archive_add_step_link_user_archive_mode_select_list_no_has_credential',
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js:50]:
```suggestion
        : 'pci_projects_project_storages_cold_archive_add_step_link_user_archive_mode_select_list_has_no_credential',
```

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+  constructor($translate, CucCloudMessage, PciStoragesColdArchiveService) {
     this.$translate = $translate;
-    this.CucCloudMessage = CucCloudMessage;
+    this.cucCloudMessage = CucCloudMessage;
+    this.pciStoragesColdArchiveService = PciStoragesColdArchiveService;
   }
 
   $onInit() {
-    this.archiveModel = { name: '' };
+    this.isArchiveCreationInprogress = false;
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js:17]:
```suggestion
    this.isArchiveCreationInProgress = false;
```

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js]: Well see 👍, done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js (hunk 1 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js]: done

File: packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js (hunk 3 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/add.controller.js]: done

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**